pkg/upgradeservice/bootstrap.go

```diff
@@ -40,6 +46,41 @@ func bootstrap(params types.UpgradeServiceParams) (finalError error) {
 	return nil
 }
 
+// updateWithinKubeRange checks if the update version is within the same major version and
+// at most one minor version ahead of the current version.
+func updateWithinKubeRange(params types.UpgradeServiceParams) error {
+	currentVersion, err := extractKubeVersion(params.CurrentECVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract current kube version")
+	}
+	updateVersion, err := extractKubeVersion(params.UpdateECVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract update kube version")
+	}
```

Comment:
We're erroring out if we fail to extract a valid kube version effectively halting the upgrade process altogether. So my question would be:
- Could there be EC releases out there without the `*+k8s-<version>*` version syntax?
- Should we be more cautious here and instead ignore the check altogether if we fail to extract any of the versions?

Reply:
new versions should all have `k8s-<version>`. i think this looks good for now.